src/main/webapp/mapScript.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let highlightedId = 0;
+
+function initMap(mapLocation) {
+  let myLatLng = {lat: mapLocation.lat, lng: mapLocation.lng};
+  map = new google.maps.Map(document.getElementById('map'), {
+    zoom: 13,
+    center: myLatLng,
+    mapTypeControl: false
+  });
+}
+
+function createMarker(listing, cardNumber) {
+  let marker = new google.maps.Marker({
+    position: {lat: listing.mapLocation.lat, lng: listing.mapLocation.lng},
+  });
+
+  marker.addListener('click', function() {
+    document.getElementById(highlightedId).style.backgroundColor = '#fff';
+    map.setCenter(marker.getPosition());
+    map.setZoom(16);
+    let firstId = document.getElementById('results-content').firstChild.id;
+    // If card is not on the current page
+    if ((cardNumber-firstId < 0) || (cardNumber-firstId >= 3)) {
+      displayCards(cardNumber-firstId);
+    }
```

Comment:
Curious. Wouldn't firstId always be 0. So unless cardNumber is also referencing the first card wouldn't this first half effectively just be true. Have you considered using a boolean to track the selected card?

Reply:
Since the results-content section only holds three children, the firstId will be the id of whatever card is first on the current page (ex. I scroll to the third page and then click a marker). The first half would be true only if the card is one of the three on the current page that the user is paginated to. Does that make sense? I decided to not use a boolean because only one card should be selected at a time, and ideally that card will only be unhighlighted when another marker is chosen to avoid confusion. An alternative would be to have the chosen card flash when the marker is clicked if you think that is a better solution then carrying over the highlighted card to the next call of the function.